lib/raw-stream-set.js

```diff
@@ -18,12 +18,29 @@ module.exports = class RawStreamSet {
     const stream = this._dht._udx.createStream(id, opts)
     this._streams.set(id, stream)
 
-    stream.on('close', onclose)
+    stream
+      .on('close', onclose)
+      .on('connect', onconnect)
 
     return stream
 
     function onclose () {
       self._streams.delete(id)
     }
+
+    async function onconnect () {
+      const {
+        remoteHost: host,
+        remotePort: port
+      } = stream
+
+      for (let mtu = 1400; mtu > 1200; mtu -= 100) {
+        try {
+          await self._dht.ping({ host, port }, { size: mtu - 12 /* overhead */ })
```

Comment:
pass the socket here also as options, stream.rawStream.socket so it has the correct session

Reply:
Terribly good point 😂